suite/tests/client-interface/drx-scattergather-aarch64.cpp

```diff
@@ -1889,8 +1923,536 @@ test_st1_scalar_plus_vector()
+            std::array<uint64_t, 2> { 0xffffffff, 0xffffffff },
+            std::array<uintptr_t, 2> {
+                get_base_ptr(element_size_t::SINGLE, 0),
+                get_base_ptr(element_size_t::SINGLE, 0),
+            },
+        },
+        {
+            "ld1sw vector+immediate 64bit element",
+            TEST_FUNC("ld1sw z19.d, p5/z, [z11.d, #20]"),
+            { /*zt=*/19, /*pg=*/5, /*zn=*/11 },
+            std::array<int64_t, 2> { 0x10, -1 },
+            std::array<uintptr_t, 2> {
+                get_base_ptr(element_size_t::SINGLE, 5),
+                get_base_ptr(element_size_t::SINGLE, 26),
+            },
+        },
+        {
+            "ld1sw vector+immediate 64bit element (max index)",
+            TEST_FUNC("ld1sw z19.d, p5/z, [z11.d, #124]"),
+            { /*zt=*/19, /*pg=*/5, /*zn=*/11 },
+            std::array<int64_t, 2> { -1, -1 },
+            std::array<uintptr_t, 2> {
+                get_base_ptr(element_size_t::SINGLE, 0),
+                get_base_ptr(element_size_t::SINGLE, 0),
+            },
+        },
+        {
+            "ld1d vector+immediate 64bit element",
+            TEST_FUNC("ld1d z23.d, p6/z, [z7.d, #48]"),
+            { /*zt=*/23, /*pg=*/6, /*zn=*/7 },
+            std::array<uint64_t, 2> { 0x12, 0x28 },
+            std::array<uintptr_t, 2> {
+                get_base_ptr(element_size_t::DOUBLE, 6),
+                get_base_ptr(element_size_t::DOUBLE, 22),
+            },
+        },
+        {
+            "ld1d vector+immediate 64bit element (max index)",
+            TEST_FUNC("ld1d z23.d, p6/z, [z7.d, #248]"),
+            { /*zt=*/23, /*pg=*/6, /*zn=*/7 },
+            std::array<uint64_t, 2> { 0xffffffffffffffff, 0xffffffffffffffff },
+            std::array<uintptr_t, 2> {
+                get_base_ptr(element_size_t::DOUBLE, 0),
+                get_base_ptr(element_size_t::DOUBLE, 0),
+            },
+        },
+        {
+            "ld1d vector+immediate 64bit element Zt==Zn",
+            TEST_FUNC("ld1d z27.d, p7/z, [z3.d, #0]"),
+            { /*zt=*/27, /*pg=*/7, /*zn=*/3 },
+            std::array<uint64_t, 2> { 0x07, 0x23 },
+            std::array<uintptr_t, 2> {
+                get_base_ptr(element_size_t::DOUBLE, 7),
+                get_base_ptr(element_size_t::DOUBLE, 23),
+            },
+        },
+    });
+#    undef TEST_FUNC
+}
+
+struct vector_plus_immediate_store_test_case_t
+    : public test_case_base_t<basic_test_ptrs_t> {
+    output_data_t output_data;
+    vector_reg_value128_t base_data;
+    std::array<const void *, 2> base_ptrs;
+
+    struct registers_used_t {
+        unsigned src_z;
+        unsigned governing_p;
+        unsigned base_z;
+    } registers_used;
+
+    element_size_t stored_value_size;
+
+    expected_values_t expected_values;
+
+    vector_plus_immediate_store_test_case_t(std::string name_, test_func_t func_,
+                                            registers_used_t registers_used_,
```

Comment:
Re these identifiers: usually we append a `_` for class/struct members, not formal arguments.

Reply:
Done.